openedx/features/badging/models.py

```diff
@@ -18,3 +24,89 @@ class Badge(models.Model):
 
     def __unicode__(self):
         return self.name
+
+    @classmethod
+    def get_remaining_badges(cls, user_id, community_id, community_type):
+        badges_in_community = UserBadge.objects.filter(user_id=user_id,
+                                                       community_id=community_id,
+                                                       badge_id__type=community_type)
+
+        if badges_in_community:
+            latest_earned = badges_in_community.latest('date_earned')
+            latest_threshold = Badge.objects.get(pk=latest_earned.badge_id, type=community_type).threshold
+            remaining_badges = Badge.objects.filter(type=community_type) \
+                                            .exclude(threshold__lte=latest_threshold) \
+                                            .order_by('threshold')
+        else:
+            remaining_badges = Badge.objects.filter(type=community_type).order_by('threshold')
+        
+        remaining_badges_json = {}
```

Comment:
Let's please remove `json` as it a dictionary.

Reply:
Done